### PR TITLE
implement ktica

### PR DESCRIPTION
**Pull request by cxhernandez**
- [x] Implement feature / fix bug
 - [ ] Add tests
 - [x] Update changelog

Third time's a charm? This is an attempt to consolidate code from https://github.com/msmbuilder/msmbuilder/pull/394, https://github.com/msmbuilder/msmbuilder/pull/696, and [conformation/LandmarkNystroem.py](https://github.com/evanfeinberg/conformation/blob/master/LandmarkNystroem.py), and make a useable implementation of kernel tICA.

### CODE REVIEW

**Comment by evanfeinberg** [comment]
+1! Very clean implementation and synthesis of LandmarkNystroem.py and #696 . In docstring for basis, maybe add "ndarray of shape (n_frames, n_features)" to specify?

File: msmbuilder/decomposition/kernel_approx.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    components_ : array, shape (n_components, n_features)
+        Subset of training points used to construct the feature map.
+    component_indices_ : array, shape (n_components)
+        Indices of ``components_`` in the training set.
+    normalization_ : array, shape (n_components, n_components)
+        Normalization matrix needed for embedding.
+        Square root of the kernel matrix on ``components_``.
+    References
+    ----------
+    * Williams, C.K.I. and Seeger, M.
```
Comment (mpharrigan) [inline msmbuilder/decomposition/kernel_approx.py]: Can you change this to do numpydoc reference style? So it shows up right in the docs

Comment (mpharrigan) [inline msmbuilder/decomposition/kernel_approx.py]: You should also add it to the docs :) Should just be a matter of listing the relevant classes in decomposition.rst

**Comment by cxhernandez** [comment]
Okay, I think this is finally ready for review. @evanfeinberg and I talked offline about perhaps making a `KernelDecomposition` base class, which would make it easier to then create new classes that use kernel approximations in the future, but maybe that can go into a new PR.

File: msmbuilder/decomposition/kernel_approximation.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
+# Author: Carlos Xavier Hernandez <cxh@stanford.edu>
+# Contributors: Muneeb Sultan <msultan@stanford.edu>, Evan Feinberg <enf@stanford.edu>
+# Copyright (c) 2015, Stanford University and the Authors
```
Comment (mpharrigan) [inline msmbuilder/decomposition/kernel_approximation.py]: 2016!

File: msmbuilder/decomposition/kernel_approximation.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+class Nystroem(MultiSequenceDecompositionMixin, kernel_approximation.Nystroem):
+    __doc__ = kernel_approximation.Nystroem.__doc__
+
+
+class LandmarkNystroem(Nystroem):
+    """Approximate a kernel map using a subset of the training data.
+
+    Constructs an approximate feature map for an arbitrary kernel
+    using a subset of the data as basis.
+    Read more in the :ref:`User Guide <nystroem_kernel_approx>`.
```
Comment (mpharrigan) [inline msmbuilder/decomposition/kernel_approximation.py]: Does this reference exist?

**Comment by mpharrigan** [comment]
Taking a quick look, lgtm. I'll try to give it a spin on a dataset in the near future but don't wait for that to merge

**Comment by cxhernandez** [comment]
Well appveyor is the real hold up 😜

File: msmbuilder/decomposition/kernel_approximation.py (hunk 3 above)
Comment (cxhernandez) [inline msmbuilder/decomposition/kernel_approximation.py]: Not exactly. @evanfeinberg mentioned that this came out of a discussion he had with @rmcgibbo, but it's also pretty much what @schwancr proposed in his ktica manuscript (except we're using the Nystroem approx. here). I did find this [reference](https://papers.nips.cc/paper/5481-fast-prediction-for-large-scale-kernel-machines.pdf) though, which suggests a similar approach (using kmeans centroids as "pseudo-landmark" points).

File: msmbuilder/decomposition/kernel_approximation.py (hunk 2 above)
Comment (cxhernandez) [inline msmbuilder/decomposition/kernel_approximation.py]: woops!

**Pull request merged**